scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+            if (uploaded) {
+                getOut().println("UPLOADED a file to: " + Paths.get(ftpRemoteFilePath, localFile.getName()));
+            } else {
+                getOut().println("COULD NOT upload the file: " + ftpLocalFilePath);
+            }
+        }
+        // If it is a folder, upload all its contents recursively
+        else if (localFile.isDirectory()) {
+            getOut().println("A DIRECTORY to upload");
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            uploadDirectory(ftpClient, "", ftpLocalFilePath, "");
+        } else {
+            throw new IllegalArgumentException(localFile + " not found. Please, enter a valid path.");
+        }
+        getOut().println("END Export file(s) to FTP.");
+    }
+
+    private void uploadDirectory(FTPClient ftpClient, String remoteDirPath, String localParentDir,
+            String remoteParentDir) throws IOException {
+
+        getOut().println("LISTING directory: " + localParentDir);
+
+        File localDir = new File(localParentDir);
+        File[] subFiles = localDir.listFiles();
+        if (subFiles != null && subFiles.length > 0) {
+            for (File item : subFiles) {
+                uploadSubFilesDirectory(ftpClient, remoteDirPath, localParentDir, remoteParentDir, item);
+            }
+        }
+    }
+
+    private void uploadSubFilesDirectory(FTPClient ftpClient, String remoteDirPath, String localParentDir,
+            String remoteParentDir, File item) throws IOException {
+        String remoteFilePath = Paths.get(remoteDirPath, remoteParentDir, item.getName()).toString();
+        if (remoteParentDir.equals("")) {
+            remoteFilePath = Paths.get(remoteDirPath, item.getName()).toString();
+        }
+
+        if (item.isFile()) {
+            // upload the file
+            String localFilePath = item.getAbsolutePath();
+            getOut().println("About to upload the file: " + localFilePath);
+            boolean uploaded = uploadSingleFile(ftpClient, localFilePath, remoteFilePath);
+            if (uploaded) {
+                getOut().println("UPLOADED a file to: " + remoteFilePath);
+            } else {
+                getOut().println("COULD NOT upload the file: " + localFilePath);
+            }
+        } else {
+            // create directory on the server
+            boolean created = ftpClient.makeDirectory(remoteFilePath);
+            if (created) {
+                getOut().println("CREATED the directory: " + remoteFilePath);
+            } else {
+                getOut().println("COULD NOT create the directory: " + remoteFilePath);
+            }
+
+            // upload the sub directory
+            String parent = Paths.get(remoteParentDir, item.getName()).toString();
+            if (remoteParentDir.equals("")) {
+                parent = item.getName();
+            }
+
+            localParentDir = item.getAbsolutePath();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "localParentDir". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1226)